src/shared/Microsoft.Git.CredentialManager/Authentication/MicrosoftAuthentication.cs

```diff
@@ -82,48 +97,61 @@ public MicrosoftAuthentication(ICommandContext context)
+                        .WithPrompt(Prompt.SelectAccount)
+                        // We must configure the system webview as a fallback
+                        .WithSystemWebViewOptions(GetSystemWebViewOptions())
+                        .ExecuteAsync();
+                }
+                else
+                {
+                    // Check for a user flow preference if they've specified one
+                    MicrosoftAuthenticationFlowType flowType = GetFlowType();
+                    switch (flowType)
+                    {
+                        case MicrosoftAuthenticationFlowType.Auto:
+                            if (CanUseEmbeddedWebView())
+                                goto case MicrosoftAuthenticationFlowType.EmbeddedWebView;
+
+                            if (CanUseSystemWebView(app, redirectUri))
+                                goto case MicrosoftAuthenticationFlowType.SystemWebView;
+
+                            // Fall back to device code flow
+                            goto case MicrosoftAuthenticationFlowType.DeviceCode;
+
+                        case MicrosoftAuthenticationFlowType.EmbeddedWebView:
+                            Context.Trace.WriteLine("Performing interactive auth with embedded web view...");
+                            EnsureCanUseEmbeddedWebView();
+                            result = await app.AcquireTokenInteractive(scopes)
+                                .WithPrompt(Prompt.SelectAccount)
+                                .WithUseEmbeddedWebView(true)
+                                .ExecuteAsync();
+                            break;
+
+                        case MicrosoftAuthenticationFlowType.SystemWebView:
+                            Context.Trace.WriteLine("Performing interactive auth with system web view...");
+                            EnsureCanUseSystemWebView(app, redirectUri);
+                            result = await app.AcquireTokenInteractive(scopes)
+                                .WithPrompt(Prompt.SelectAccount)
+                                .WithSystemWebViewOptions(GetSystemWebViewOptions())
+                                .ExecuteAsync();
+                            break;
+
+                        case MicrosoftAuthenticationFlowType.DeviceCode:
+                            Context.Trace.WriteLine("Performing interactive auth with device code...");
+                            // We don't have a way to display a device code without a terminal at the moment
+                            // TODO: introduce a small GUI window to show a code if no TTY exists
+                            ThrowIfTerminalPromptsDisabled();
+                            result = await app.AcquireTokenWithDeviceCode(scopes, ShowDeviceCodeInTty).ExecuteAsync();
+                            break;
+
+                        default:
+                            goto case MicrosoftAuthenticationFlowType.Auto;
+                    }
```

Comment:
The indentation change makes this harder to review than I'd like, but that cannot be helped, I guess.

Reply:
Yeah. This is unfortunate 😢

Reply:
![image](https://user-images.githubusercontent.com/570044/114253834-fd9dc580-9979-11eb-8132-faa0b197fc8b.png)

Reloading to ignore whitespace changes works great!

![image](https://user-images.githubusercontent.com/570044/114253836-02627980-997a-11eb-9805-624f5498e5eb.png)